lib/block_view_validator.go

```diff
@@ -592,83 +629,63 @@ func DBGetValidatorByPKIDWithTxn(txn *badger.Txn, snap *Snapshot, pkid *PKID) (*
 	return validatorEntry, nil
 }
 
-func DBGetTopActiveValidatorsByStake(
+func DBGetTopActiveValidatorsByStakeAmount(
 	handle *badger.DB,
 	snap *Snapshot,
 	limit uint64,
 	validatorEntriesToSkip []*ValidatorEntry,
 ) ([]*ValidatorEntry, error) {
 	var validatorEntries []*ValidatorEntry
 
-	// Convert ValidatorEntriesToSkip to ValidatorEntryKeysToSkip.
-	validatorKeysToSkip := NewSet([]string{})
+	// Convert validatorEntriesToSkip to the ValidatorPKIDs we need to skip.
+	validatorPKIDsToSkip := NewSet([]PKID{})
 	for _, validatorEntryToSkip := range validatorEntriesToSkip {
-		validatorKeysToSkip.Add(string(DBKeyForValidatorByStake(validatorEntryToSkip)))
```

Comment:
There was a subtle issue here in cases where a validator had its TotalStakeAmountNanos field updated in the view. Since the TotalStakeAmountNanos field is used in the DB key and is a mutable value, it is unreliable to use for filtering during DB seeks. 

This seek would have returned a validators that were intended to be filtered out.

Reply:
Oh wow good catch.